src/geo/map-view-factory.js

```diff
@@ -0,0 +1,33 @@
+/* global google */
+var cdb = require('cdb');
+var LeafletLayerViewFactory = require('./leaflet/leaflet-layer-view-factory');
+var GMapsLayerViewFactory = require('./gmaps/gmaps-layer-view-factory');
+
+var MapViewFactory = function () {};
+
+MapViewFactory.prototype.createMapView = function (provider, mapModel, el) {
+  var MapViewClass;
+  var LayerViewFactoryClass;
+
+  if (provider === 'leaflet') {
+    MapViewClass = cdb.geo.LeafletMapView;
+    LayerViewFactoryClass = LeafletLayerViewFactory;
+  } else if (provider === 'googlemaps') {
+    if (typeof (google) !== 'undefined' && typeof (google.maps) !== 'undefined') {
+      MapViewClass = cdb.geo.GoogleMapsMapView;
+      LayerViewFactoryClass = GMapsLayerViewFactory;
+    } else {
+      throw new Error('Google maps library should be included');
+    }
+  } else {
+    throw new Error(provider + ' provider is not supported');
+  }
+
+  return new MapViewClass({
+    el: el,
+    map: mapModel,
+    layerViewFactory: new LayerViewFactoryClass()
```

Comment:
don't totally understand why the factory should be passed but other things are done though inheritance. Does make sense to change the factory for a map implementation?